AirCasting/SearchAndFollow/SearchMap/BottomCardViewModel.swift

```diff
@@ -0,0 +1,47 @@
+// Created by Lunar on 04/04/2022.
+//
+
+import Foundation
+
+class BottomCardViewModel: ObservableObject {
+    @Published private var isModalScreenPresented = false
+    let dataModel: BottomCardModel
+    
+    init(id: Int, title: String, startTime: String, endTime: String, latitude: Double, longitude: Double) {
+        dataModel = .init(id: id, title: title, startTime: startTime, endTime: endTime, latitude: latitude, longitude: longitude)
+    }
+    
+    func getIsModalScreenPresented() -> Bool { isModalScreenPresented }
+    func setIsModalScreenPresented() { isModalScreenPresented.toggle() }
+    
+    func adaptTimeAndDate() -> String {
+        let formatter = DateFormatters.SessionCartView.utcDateIntervalFormatter
+        let start = startTimeAsDate()
+        let end = endTimeAsDate()
+        let string = formatter.string(from: start, to: end)
+        return string
+    }
+    
+    func startTimeAsDate() -> Date {
+        let formatter = DateFormatters.SearchAndFollow.timeFormatter
+        let date = formatter.date(from: dataModel.startTime)
+        guard let d = date else { return DateBuilder.getFakeUTCDate() }
+        return d
+    }
+    
+    func endTimeAsDate() -> Date {
+        let formatter = DateFormatters.SearchAndFollow.timeFormatter
+        let date = formatter.date(from: dataModel.endTime)
+        guard let d = date else { return DateBuilder.getFakeUTCDate() }
+        return d
+    }
+    
+    func initCompleteScreen() -> CompleteScreen {
+        CompleteScreen(session: .init(uuid: .init(rawValue: "\(dataModel.id)") ?? .init(),
+                                      name: dataModel.title,
+                                      startTime: startTimeAsDate(),
+                                      endTime: endTimeAsDate(),
+                                      longitude: dataModel.longitude,
+                                      latitude: dataModel.latitude))
```

Comment:
Probably we can add some dismiss function here to fix the `X` mark. Or we can delete the `X` mark, cause it doesn't really look so good. I included it in the complete screen cause it was present in the designs

Reply:
I think it does look ok here. I do only use swipe-down gestures but who knows about others. I will fix it, and if we could leave it, it would be great